File: src/main/java/io/myosotisdev/minestom/module/PermissionModule.java
package io.myosotisdev.minestom.module;

import net.minestom.server.command.CommandSender;
import net.minestom.server.entity.Player;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

public class PermissionModule
        extends Module
{
    private Set<UUID> operators;

    public PermissionModule()
    {
        super("permission");
        this.operators = new HashSet<>();
    }

    public void setOp(Player player, boolean enable)
    {
        if(player == null)
            return;

        if(enable)
        {
            operators.add(player.getUuid());
            player.sendMessage("You have been set to operator.");
        }
        else
        {
            operators.remove(player.getUuid());
            player.sendMessage("You are not operator any more");
        }
    }

    public boolean isOp(CommandSender sender)
    {
        return (sender instanceof Player) && isOp((Player) sender);
    }

    public boolean isOp(Player player)
    {
        return operators.contains(player.getUuid());
    }

    @Override
    public void onEnable()
    {

    }

    @Override
    public void onDisable()
    {

    }
}

File: src/main/java/io/myosotisdev/minestom/module/PermissionModule.kt
package io.myosotisdev.minestom.module

import io.myosotisdev.minestom.permission.PermissionGroup
import net.minestom.server.command.CommandSender
import net.minestom.server.entity.Player
import java.util.*
import kotlin.collections.HashSet

class PermissionModule : Module("permission")
{
    private val operators: MutableSet<UUID> = HashSet()
    val groups: MutableSet<PermissionGroup> = HashSet()

    fun setOp(player: Player?, enable: Boolean)
    {
        if (player == null) return
        if (enable)
        {
            operators.add(player.uuid)
            player.sendMessage("You have been set to operator.")
        }
        else
        {
            operators.remove(player.uuid)
            player.sendMessage("You are not operator any more")
        }
    }

    fun isOp(sender: CommandSender?): Boolean
    {
        return sender is Player && isOp(sender)
    }

    fun isOp(player: Player): Boolean
    {
        return operators.contains(player.uuid)
    }

    override fun onEnable()
    {
    }

    override fun onDisable()
    {
    }
}